File: src/Symfony/Component/EventDispatcher/EventSubscriberInterface.php
<?php

/*
 * This file is part of the Symfony package.
 *
 * (c) Fabien Potencier <fabien@symfony.com>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

namespace Symfony\Component\EventDispatcher;

/**
 * An EventSubscriber knows himself what events he is interested in.
 * If an EventSubscriber is added to an EventDispatcherInterface, the manager invokes
 * {@link getSubscribedEvents} and registers the subscriber as a listener for all
 * returned events.
 *
 * @link    www.doctrine-project.org
 * @since   2.0
 * @author  Guilherme Blanco <guilhermeblanco@hotmail.com>
 * @author  Jonathan Wage <jonwage@gmail.com>
 * @author  Roman Borschel <roman@code-factory.org>
 * @author  Bernhard Schussek <bschussek@gmail.com>
 *
 * @api
 */
interface EventSubscriberInterface
{
    /**
     * Returns an array of event names this subscriber wants to listen to.
     *
     * The array keys are event names and the value can be:
     *
     *  * The method name to call (priority defaults to 0)
     *  * An array composed of the method name to call and the priority
     *
     * For instance:
     *
     *  * array('eventName' => 'methodName')
     *  * array('eventName' => array('methodName', $priority))
     *
     * @return array The event names to listen to
     *
     * @api
     */
    static function getSubscribedEvents();
}

File: src/Symfony/Component/EventDispatcher/Resources/ext/event_subscriber_interface.c
/*
 * This file is part of the Symfony package.
 *
 * (c) Fabien Potencier <fabien@symfony.com>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

#include "event_dispatcher_main.h"

zend_class_entry *event_subscriber_interface_ce;
static zend_function_entry event_subscriber_interface_methods[];

PHP_MINIT_FUNCTION(event_subscriber_interface)
{
  zend_class_entry ce;
  INIT_CLASS_ENTRY(ce, ZEND_NS_NAME(EVENT_DISPATCHER_NS, "EventSubscriberInterface"), event_subscriber_interface_methods);
  event_subscriber_interface_ce = zend_register_internal_interface(&ce TSRMLS_CC);

  return SUCCESS;
}

static zend_function_entry event_subscriber_interface_methods[] = {
    ZEND_FENTRY(getSubscribedEvents, NULL, NULL, ZEND_ACC_STATIC|ZEND_ACC_PUBLIC|ZEND_ACC_ABSTRACT)
    PHP_FE_END
};
